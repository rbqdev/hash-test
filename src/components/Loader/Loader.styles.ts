import styled from "styled-components";
import { opacify } from "polished";

export const Wrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;

  &:before {
    content: "";
    display: block;
    width: 100%;
    height: 100%;
    background: rgb(246 247 251 / 20%);
    border-radius: 4px;
    position: absolute;
    z-index: 1;
    order: 1;
  }

  &:after {
    content: "";
    display: block;
    width: 50px;
    height: 50px;
    margin: 16px;
    border-radius: 50%;
    border: 4px solid ${({ theme }) => theme.colors.blueLight};
    border-color: ${({ theme }) => theme.colors.blueLight} transparent
      ${({ theme }) => theme.colors.blueLight} transparent;
    animation: loaderAnimation 1s linear infinite;
    position: relative;
    z-index: 2;
  }

  @keyframes loaderAnimation {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
`;

export const LoaderText = styled.span`
  color: ${({ theme }) => theme.colors.white};
  background: ${({ theme }) => opacify(0.5, theme.colors.black)};
  white-space: nowrap;
  order: 2;
  padding: 4px 8px;
  border-radius: 2px;
  position: absolute;
  bottom: 30%;
`;
