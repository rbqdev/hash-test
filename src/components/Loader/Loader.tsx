import { ScreenReaderText } from "@styles/global";
import * as Styled from "./Loader.styles";

interface LoaderProps {
  customText?: string;
}

export default function Loader({ customText = "" }: LoaderProps) {
  return (
    <Styled.Wrapper role="status">
      <ScreenReaderText>Carregando...</ScreenReaderText>
      {customText && <Styled.LoaderText>{customText}</Styled.LoaderText>}
    </Styled.Wrapper>
  );
}
