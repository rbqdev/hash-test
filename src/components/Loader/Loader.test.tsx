import render from "@testsUtils/renderWithTheme";
import Loader from ".";

describe("Component: Loader", () => {
  it("should render component correctly", () => {
    const { baseElement } = render(<Loader />);
    expect(baseElement).toMatchSnapshot();
  });

  describe("if customText", () => {
    const mockText = "texTest";

    it("should render if passed", () => {
      const { queryByText } = render(<Loader customText={mockText} />);

      expect(queryByText(mockText)).toBeInTheDocument();
    });

    it("should NOT render if not passed", () => {
      const { queryByText } = render(<Loader />);

      expect(queryByText(mockText)).not.toBeInTheDocument();
    });
  });
});
