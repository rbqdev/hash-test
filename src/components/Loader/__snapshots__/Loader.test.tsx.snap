// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Loader should render component correctly 1`] = `
.c1 {
  border: 0;
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
  word-wrap: normal !important;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
}

.c0:before {
  content: "";
  display: block;
  width: 100%;
  height: 100%;
  background: rgb(246 247 251 / 20%);
  border-radius: 4px;
  position: absolute;
  z-index: 1;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
}

.c0:after {
  content: "";
  display: block;
  width: 50px;
  height: 50px;
  margin: 16px;
  border-radius: 50%;
  border: 4px solid #5d9cec;
  border-color: #5d9cec transparent #5d9cec transparent;
  -webkit-animation: loaderAnimation 1s linear infinite;
  animation: loaderAnimation 1s linear infinite;
  position: relative;
  z-index: 2;
}

<body>
  <div>
    <div
      class="c0"
      role="status"
    >
      <div
        class="c1"
      >
        Carregando...
      </div>
      
    </div>
  </div>
</body>
`;
